<launch>

  <!-- Map server -->
  <arg name="map_file" default="$(find chefbot_bringup)/map/map_bureau_v4.yaml"/>

  <!-- Localization -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />
  <arg name="use_map_topic"  default="false"/>
  <arg name="scan_topic"     default="scan"/>
  <arg name="initial_pose_x" default="0.0"/>
  <arg name="initial_pose_y" default="0.0"/>
  <arg name="initial_pose_a" default="0.0"/>
  <arg name="initial_cov_xx" default="0.001"/>
  <arg name="initial_cov_yy" default="0.001"/>
  <arg name="initial_cov_aa" default="0.00001"/>

  <node pkg="amcl" type="amcl" name="amcl">
  <rosparam file="$(find my_amcl_launcher)/params/my_amcl_params.yaml" command="load"/> 
    <param name="odom_model_type"           value="diff"/>    
    <param name="use_map_topic"             value="$(arg use_map_topic)"/>
    <param name="odom_frame_id"             value="odom"/>
    <param name="laser_model_type"          value="likelihood_field"/>
    <param name="base_frame_id"             value="base_footprint"/>
    <param name="initial_pose_x"            value="$(arg initial_pose_x)"/>
    <param name="initial_pose_y"            value="$(arg initial_pose_y)"/>
    <param name="initial_pose_a"            value="$(arg initial_pose_a)"/>
    <remap from="scan"                      to="$(arg scan_topic)"/>
  </node>

  <!--- Run Move Base -->
  <include file="$(find chefbot_bringup)/launch/includes/move_base.launch.xml"/>

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find chefbot_bringup)/rviz/config_move_base.rviz" />

</launch>